packages/plugin-core/src/commands/OpenLink.ts

```diff
@@ -25,47 +27,59 @@ export class OpenLinkCommand extends BasicCommand<CommandOpts, CommandOutput> {
   async execute() {
     const ctx = DENDRON_COMMANDS.OPEN_LINK;
     this.L.info({ ctx });
+
+    let text = "";
+
     if (!isAnythingSelected()) {
-      const error = DendronError.createFromStatus({
-        status: ERROR_STATUS.INVALID_STATE,
-        message: `nothing selected`,
-      });
-      this.L.error({ error });
-      return { error };
+      window.showInformationMessage(
+        "nothing selected, searching for valid link"
+      );
     }
-    const { text } = VSCodeUtils.getSelection();
-    if (_.isUndefined(text)) {
+
+    text = getURLAt(VSCodeUtils.getActiveTextEditor());
+
+    if (_.isUndefined(text) || text === "") {
       const error = DendronError.createFromStatus({
         status: ERROR_STATUS.INVALID_STATE,
-        message: `nothing selected`,
+        message: `no valid path or URL selected`,
       });
       this.L.error({ error });
       return { error };
     }
-    const wsRoot = getDWorkspace().wsRoot;
+
     let assetPath: string;
-    if (text.startsWith("asset")) {
-      const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
-      assetPath = path.join(vault2Path({ vault, wsRoot }), text);
+    if (text.indexOf("://") !== -1) {
+      window.showInformationMessage(
+        "the selection reads as a full URI so an attempt will be made to open it"
+      );
+      env.openExternal(Uri.parse(text));
+      assetPath = text;
     } else {
-      assetPath = resolvePath(text, getExtension().rootWorkspace.uri.fsPath);
-    }
-    if (!fs.existsSync(assetPath)) {
-      const error = DendronError.createFromStatus({
-        status: ERROR_STATUS.INVALID_STATE,
-        message: `${assetPath} does not exist`,
+      const wsRoot = getDWorkspace().wsRoot;
+
+      if (text.startsWith("asset")) {
+        const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
+        assetPath = path.join(vault2Path({ vault, wsRoot }), text);
+      } else {
+        assetPath = resolvePath(text, getExtension().rootWorkspace.uri.fsPath);
+      }
+      if (!fs.existsSync(assetPath)) {
+        const error = DendronError.createFromStatus({
+          status: ERROR_STATUS.INVALID_STATE,
+          message: `no valid path or URL selected`,
+        });
+        this.L.error({ error });
+        return { error };
+      }
+      await open(assetPath).catch((err) => {
+        const error = DendronError.createFromStatus({
+          status: ERROR_STATUS.UNKNOWN,
+          error: err,
+        });
+        this.L.error({ error });
+        return { error };
       });
-      this.L.error({ error });
-      return { error };
     }
-    await open(assetPath).catch((err) => {
-      const error = DendronError.createFromStatus({
-        status: ERROR_STATUS.UNKNOWN,
-        error: err,
-      });
-      this.L.error({ error });
-      return { error };
-    });
-    return { filePath: assetPath };
+    return { String: assetPath };
```

Comment:
avoid generic names like `String`. unless there's good reason, keep the existing name `filePath`